src/metabase/lib/field.cljc

```diff
@@ -787,15 +787,16 @@
 (mu/defn ^:private remapped-field :- [:maybe ::lib.schema.metadata/column]
```

Comment:
PK values are not remapped to type/Name fields. But we allow searching by Name. That's why `remapped-field` checks for FKs only.